feature_engine/variable_manipulation.py

```diff
@@ -2,33 +2,77 @@
 # License: BSD 3 clause
 # functions shared across transformers
 
+from typing import List, Optional, Union
 
-def _define_variables(variables):
-    # Check that variable names are passed in a list.
-    # Can take None as value
+import pandas as pd
+
+
+def _define_variables(variables: Union[None, List[str]]) -> Union[List[str], None]:
+    """
+    Takes string or list of strings and checks if argument is list of strings.
+    Can take None as argument.
+
+    Args:
+        variables: string or list of strings
+
+    Returns:
+        List of strings
+    """
+
+    #! Can take Tuple of variables and return list of tuple. Is it accaptable?
     if not variables or isinstance(variables, list):
         variables = variables
     else:
         variables = [variables]
+
     return variables
 
 
-def _define_numerical_dict(the_dict):
-    # Check that the entered dictionary is indeed a dictionary of integers and floats
-    # Can take None as value
+def _define_numerical_dict(the_dict: Union[Optional[str], dict]) -> Union[Optional[str], dict]:
+    """
```

Comment:
the_dict can only take None or a dictionary. Is this what the Union argument is saying? It shouldn't take a string.

also, is `the_dict` a good name for this param? I think we should use `dict_`.

I named it on a rush when we were working on this function.

Reply:
In that case I think we want `Optional[dict]`